packages/neutrino/bin/start.js

```diff
@@ -1,31 +1,57 @@
 const { start } = require('../src');
 const ora = require('ora');
+const dns = require('dns');
+const os = require('os');
+
+const platformHostName = os.hostname();
+const whenIPReady = new Promise((done, failed) => {
+  dns.lookup(platformHostName, (err, ip) => {
+    if (err) {
+      failed(err);
+    } else {
+      done(ip);
+    }
+  });
+});
+
+whenIPReady.then(ip => process.env.HOST = process.env.HOST || ip);
 
 module.exports = (middleware, options) => {
   const spinner = ora('Building project').start();
 
-  return start(middleware, options)
-    .fork((errors) => {
-      spinner.fail('Building project failed');
-      errors.forEach(err => console.error(err));
-      process.exit(1);
-    }, (compiler) => { // eslint-disable-line consistent-return
-      if (!compiler.options.devServer) {
-        return spinner.succeed('Build completed');
-      }
+  return whenIPReady.then(ip => start(middleware, options).fork(
+      (errors) => {
+        spinner.fail('Building project failed');
+        errors.forEach(err => console.error(err));
 
-      const { devServer } = compiler.options;
-      const protocol = devServer.https ? 'https' : 'http';
-      const { host, port } = devServer;
+        process.exit(1);
+      },
+      (compiler) => {
+        if (!compiler.options.devServer) {
+          return spinner.succeed('Build completed');
+        }
 
-      spinner.succeed(`Development server running on: ${protocol}://${host}:${port}`);
+        const { devServer } = compiler.options;
+        const protocol = devServer.https ? 'https' : 'http';
+        const { port } = devServer;
+        let { host } = devServer;
```

Comment:
If we move the IP work to devServer.js, we shouldn't need to change anything else after here.